lib/utils/apiFunction.ts

```diff
@@ -44,7 +44,8 @@ export function registerAPIEndpoint (
     stack: Stack,
     api: IRestApi,
     authorizer: IAuthorizer,
-    appRole: IRole,
+    role: IRole,
+    appRoleName: string,
```

Comment:
I won't block on this but it feels a little unnecessary. We already have MLSpaceConfig being passed in, which has `APP_ROLE_ARN` on it and the name can be extracted from that right? Seems like a lot of changes to pass something in a different way when we could just derive it with what we have.

Reply:
If this is true I feel like it's worth blocking on... no reason to add more complexity and make this harder for the guy person to understand/maintain.

Reply:
`APP_ROLE_ARN` isn't guaranteed. It is only added if the customer is manually managing IAM roles